generics/services/survey.js

```diff
@@ -5,757 +5,617 @@
  * Description : Assessment service related information.
  */
 
- //dependencies
-const request = require('request');
+//dependencies
+const request = require('request')
 
-const ASSESSMENT_URL = process.env.ML_SURVEY_SERVICE_URL;
+const SURVEY_SERVICE_URL = process.env.SURVEY_SERVICE_URL
 
 /**
-  * Assessment create
-  * @function
-  * @name createAssessmentSolutionFromTemplate
-  * @param {String} token - logged in user token.
-  * @param {String} templateId - template id.
-  * @param {Object} bodyData - Body data
-  * @returns {JSON} - Create assessment from template solution.
-*/
-
-const createAssessmentSolutionFromTemplate = function (token,templateId,bodyData) {
-    return new Promise(async (resolve, reject) => {
-        try {
-
-            let assessmentCreateUrl =  
-            ASSESSMENT_URL + 
-            CONSTANTS.endpoints.ASSESSMENTS_CREATE + "?solutionId=" + templateId;
-
-            const options = {
-                headers : {
-                    "content-type": "application/json",
-                    "internal-access-token": process.env.INTERNAL_ACCESS_TOKEN,
-                    "x-authenticated-user-token" : token,
-                },
-                json : bodyData
-            };
-            
-            request.post(assessmentCreateUrl,options,assessmentCallback);
-
-            function assessmentCallback(err, data) {
-
-                let result = {
-                    success : true
-                };
-                
-                if (err) {
-                    result.success = false;
-                } else {
-
-                    let response = data.body;
-                    if( response.status === HTTP_STATUS_CODE['ok'].status ) {
-                        result["data"] = response.result;
-                    } else {
-                        result.success = false;
-                    }
-                }
-                return resolve(result);
-            }
-
-        } catch (error) {
-            return reject(error);
-        }
-    })
+ * Create Child solutions for survey
+ * @function
+ * @name importSurveyTemplateToSolution
+ * @param {String} token - logged in user token.
+ * @param {String} solutionId - parent solution id.
+ * @param {String} programId -programId
+ * @param {Object} data - Body data
+ * @returns {JSON} - Create child solution from parent  solution.
+ */
+const importSurveyTemplateToSolution = function (token, solutionId, programId, data = {}, userDetails) {
```

Comment:
@praveenKDass combine all creation into 1 function with a switch case if needed. Also, check and ensure, program id is mandatory else return error in these functions.